### PR TITLE
Add Vue support

### PR DESCRIPTION
**Pull request by isuda**
- Update dependencies in package.json
- add dependencies for rollup: typescript, vue...... and commonjs (🤦🏻‍♂️: https://stackoverflow.com/questions/54379726/rollup-plugin-vue-v4-6-2-has-a-problem-with-vue-runtime-helpers)
- add Vue component equivalent of the angular component.
- added a top level tsconfig.json file (same as es6 version) for use in vscode / other editors
- updated README with new instructions for Vue usage.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -91,6 +90,30 @@ Use the `ay-menu-button/dist/angular1.js` file:
 ```
 
 
+### Vue
+
+Use the `ay-menu-button/dist/vue.js` file:
+
+```javascript
+  import Vue from 'vue';
+  import AyMenuButton from 'ay-menu-button/vue';
```
Comment (dpogue) [inline README.md]: I think this needs to be `ay-menu-button/dist/vue`?

File: src/manager.ts [hunk 2]
```diff
@@ -331,48 +331,48 @@ export class MenuManager {
...
+            return doc!.scrollingElement.scrollTop;
         } else {
-            return doc.documentElement.scrollTop + doc.body.scrollTop;
+            return doc!.documentElement.scrollTop + doc!.body.scrollTop;
         }
     }
 
 
     private static blockScrolling(offset : number) {
         let doc = this.curButton!.ownerDocument;
```
Comment (dpogue) [inline src/manager.ts]: Are we able to do `let doc = this.curButton!.ownerDocument!` and then avoid adding `!` to doc everywhere below?

Comment (isuda) [reply src/manager.ts]: updated!

File: README.md (hunk 1 above)
Comment (isuda) [reply README.md]: updated!

**Pull request merged**